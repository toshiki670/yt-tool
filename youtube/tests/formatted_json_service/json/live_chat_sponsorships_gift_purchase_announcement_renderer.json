{
  "clickTrackingParams": "CAEQl98BIhMIz5X5ur27igMVoc80Bx03uTj5",
  "replayChatItemAction": {
    "actions": [
      {
        "clickTrackingParams": "CAEQl98BIhMIz5X5ur27igMVoc80Bx03uTj5",
        "addChatItemAction": {
          "item": {
            "liveChatSponsorshipsGiftPurchaseAnnouncementRenderer": {
              "id": "ChwKGkNJZk5wTW1vdTRvREZYckV3Z1FkalRvN01B",
              "timestampUsec": "1734868455220457",
              "authorExternalChannelId": "UCU_76ukhKKzBOdO6MaijWrw",
              "header": {
                "liveChatSponsorshipsHeaderRenderer": {
                  "authorName": { "simpleText": "投稿者名" },
                  "authorPhoto": {
                    "thumbnails": [
                      {
                        "url": "https://yt4.ggpht.com/ZASbqQwRUQaQitO8mNUKqjDNOonVH2swYzyFpUaF63YXbddW7MEerd_CO23NRc0JmVvBMV9h=s32-c-k-c0x00ffffff-no-rj",
                        "width": 32,
                        "height": 32
                      },
                      {
                        "url": "https://yt4.ggpht.com/ZASbqQwRUQaQitO8mNUKqjDNOonVH2swYzyFpUaF63YXbddW7MEerd_CO23NRc0JmVvBMV9h=s64-c-k-c0x00ffffff-no-rj",
                        "width": 64,
                        "height": 64
                      }
                    ]
                  },
                  "primaryText": {
                    "runs": [
                      { "text": "Gifted ", "bold": true },
                      { "text": "5", "bold": true },
                      { "text": " ", "bold": true },
                      { "text": "channel name", "bold": true },
                      { "text": " memberships", "bold": true }
                    ]
                  },
                  "authorBadges": [
                    {
                      "liveChatAuthorBadgeRenderer": {
                        "customThumbnail": {
                          "thumbnails": [
                            {
                              "url": "https://yt3.ggpht.com/c9FVnU3zQyRhy7W4z_vm4hGF7bjSyXcyxIVTELl9dDNndD-tXfDNCjRQxHfgZu-NCHq1j4VmEQ=s16-c-k",
                              "width": 16,
                              "height": 16
                            },
                            {
                              "url": "https://yt3.ggpht.com/c9FVnU3zQyRhy7W4z_vm4hGF7bjSyXcyxIVTELl9dDNndD-tXfDNCjRQxHfgZu-NCHq1j4VmEQ=s32-c-k",
                              "width": 32,
                              "height": 32
                            }
                          ]
                        },
                        "tooltip": "Member (1 year)",
                        "accessibility": {
                          "accessibilityData": { "label": "Member (1 year)" }
                        }
                      }
                    }
                  ],
                  "contextMenuEndpoint": {
                    "clickTrackingParams": "CKkCENzDCiITCM-V-bq9u4oDFaHPNAcdN7k4-Q==",
                    "commandMetadata": {
                      "webCommandMetadata": { "ignoreNavigation": true }
                    },
                    "liveChatItemContextMenuEndpoint": {
                      "params": "Q2g0S0hBb2FRMGxtVG5CTmJXOTFORzlFUmxoeVJYZG5VV1JxVkc4M1RVRWFLU29uQ2hoVlExcHNSRmg2UjI5dk4yUTBOR0ozWkU1UFlrWmhZMmNTQzBsb1NsY3RXVkJLTFhvd0lBRW9BVElhQ2hoVlExVmZOeloxYTJoTFMzcENUMlJQTmsxaGFXcFhjbmM0QWtnQlVDUSUzRA=="
                    }
                  },
                  "contextMenuAccessibility": {
                    "accessibilityData": { "label": "Chat actions" }
                  },
                  "image": {
                    "thumbnails": [
                      {
                        "url": "https://www.gstatic.com/youtube/img/sponsorships/sponsorships_gift_purchase_announcement_artwork.png"
                      }
                    ]
                  }
                }
              }
            }
          },
          "clientId": "CIfNpMmou4oDFXrEwgQdjTo7MA"
        }
      }
    ],
    "videoOffsetTimeMsec": "0"
  }
}
