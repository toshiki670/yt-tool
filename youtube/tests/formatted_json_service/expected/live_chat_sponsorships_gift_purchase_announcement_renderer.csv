id,authorExternalChannelId,postedAt,category,authorName,content
ChwKGkNJZk5wTW1vdTRvREZYckV3Z1FkalRvN01B,UCU_76ukhKKzBOdO6MaijWrw,2024-12-22T20:54:15.220457+09:00,Sponsorships gift purchase announcement,„Åª,message: 'Gifted 5 channel name memberships'
